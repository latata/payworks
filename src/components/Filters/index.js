import React from 'react';

function Filters({ organization, onOrganizationChange, language, onLanguageChange, languages }) {
  return (
    <div className="row">
      <div className="form-group col-sm">
        <label htmlFor="organization">Organization</label>
        <input
          type="text"
          className="form-control"
          id="organization"
          value={organization}
          onChange={(event) => onOrganizationChange(event.target.value)} />
      </div>
      <div className="form-group col-sm">
        <label htmlFor="language">Language</label>
        <select
          className="form-control"
          id="language"
          value={language}
          onChange={(event) => onLanguageChange(event.target.value)}>
          {languages.map(lang => <option key={lang} value={lang}>{lang || 'ALL'}</option>)}
        </select>
      </div>
    </div>
  )
}

export default Filters;
