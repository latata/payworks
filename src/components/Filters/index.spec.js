import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import Filters from './index';

test('Filters component renders and reacts to input updates', () => {
  const onLanguageChange = jest.fn();
  const onOrganizationChange = jest.fn();

  const { asFragment, getByLabelText } = render(
    <Filters
      organization="test-org"
      language="JavaScript"
      languages={['JavaScript', 'PHP', 'Java']}
      onLanguageChange={onLanguageChange}
      onOrganizationChange={onOrganizationChange}
    />
  );

  expect(asFragment()).toMatchSnapshot();

  fireEvent.change(getByLabelText('Organization'), {
    target: { value: 'new-org' }
  });

  expect(onOrganizationChange).toBeCalledWith('new-org');

  fireEvent.change(getByLabelText('Language'), {
    target: { value: 'Java' }
  });

  expect(onLanguageChange).toBeCalledWith('Java');
});
