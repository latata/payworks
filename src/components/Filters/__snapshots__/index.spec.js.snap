// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component renders and reacts to input updates 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="form-group col-sm"
    >
      <label
        for="organization"
      >
        Organization
      </label>
      <input
        class="form-control"
        id="organization"
        type="text"
        value="test-org"
      />
    </div>
    <div
      class="form-group col-sm"
    >
      <label
        for="language"
      >
        Language
      </label>
      <select
        class="form-control"
        id="language"
      >
        <option
          value="JavaScript"
        >
          JavaScript
        </option>
        <option
          value="PHP"
        >
          PHP
        </option>
        <option
          value="Java"
        >
          Java
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
